### PR TITLE
OC-359_Create_a_core_webmethod_to_change_the_status_of_a_device

### PR DESCRIPTION
**Pull request by riannefijn**

### CODE REVIEW

**Comment by MartijnSipsCGI** [comment]
retest this please

**Comment by robindenadel** [comment]
solve conflicts

File: cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/glue/steps/ws/core/devicemanagement/DeviceLifecycleStatusSteps.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        final Device device = this.DeviceRepository
+                .findByDeviceIdentification(settings.get(PlatformKeys.KEY_DEVICE_IDENTIFICATION));
+
+        assertEquals("Device Lifecycle Status should be " + deviceLifecycleStatus,
+                com.alliander.osgp.domain.core.valueobjects.DeviceLifecycleStatus.valueOf(deviceLifecycleStatus),
+                device.getDeviceLifecycleStatus());
+    }
+
+    @Then("^the status change is logged in the audit trail$")
+    public void theStatusChangeIsLoggedInTheAuditTrail(final Map<String, String> settings) throws Throwable {
```
Comment (robindenadel) [inline cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/glue/steps/ws/core/devicemanagement/DeviceLifecycleStatusSteps.java]: remove this method

File: cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/glue/steps/ws/core/devicemanagement/DeviceLifecycleStatusSteps.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import com.alliander.osgp.domain.core.repositories.DeviceRepository;
+import com.alliander.osgp.logging.domain.entities.DeviceLogItem;
+import com.alliander.osgp.logging.domain.repositories.DeviceLogItemRepository;
+
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+
+public class DeviceLifecycleStatusSteps {
+
+    private static final String PATTERN_RETRY_OPERATION = "retry count= .*, correlationuid= .*";
```
Comment (robindenadel) [inline cucumber-tests-platform-common/src/test/java/com/alliander/osgp/cucumber/platform/common/glue/steps/ws/core/devicemanagement/DeviceLifecycleStatusSteps.java]: remove this constant

File: cucumber-tests-platform-microgrids/src/test/resources/features/osgp-adapter-ws-core/DeviceLifeCycle.feature [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  Scenario Outline: Set rtu device status
+    Given an rtu iec61850 device
+      | DeviceIdentification | RTU10001 |
+    When the SetDeviceLifecycleStatus request is received
+      | DeviceIdentification  | RTU10001 |
+      | DeviceLifecycleStatus | <Status> |
+    Then the device lifecycle status in the database is
+      | DeviceIdentification  | RTU10001 |
+      | DeviceLifecycleStatus | <Status>          |
+    And the status change is logged in the audit trail
```
Comment (robindenadel) [inline cucumber-tests-platform-microgrids/src/test/resources/features/osgp-adapter-ws-core/DeviceLifeCycle.feature]: remove this step

File: cucumber-tests-platform-microgrids/src/test/resources/features/osgp-adapter-ws-core/DeviceLifeCycle.feature [hunk 4]
```diff
@@ -0,0 +1,30 @@
+@Microgrids @Platform
+Feature: Core Operations, DeviceLifeCycle
+  As a grid operator
+  I want to distinguish the various statuses of a device
+  So I know what I can or cannot do with the device
+
+# This skip tag can be removed as soon as all code from shared and platform is merged and deployed.
+  @Skip
```
Comment (robindenadel) [inline cucumber-tests-platform-microgrids/src/test/resources/features/osgp-adapter-ws-core/DeviceLifeCycle.feature]: remove comment and @Skip

File: cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/DeviceManagement/DeviceLifeCycle.feature [hunk 5]
```diff
@@ -0,0 +1,30 @@
+@PublicLighting @Platform @CoreDeviceManagement
+Feature: Core Operations, DeviceLifeCycle
+  As a grid operator
+  I want to distinguish the various statuses of a device
+  So I know what I can or cannot do with the device
+
+# This skip tag can be removed as soon as all code from shared and platform is merged and deployed.
+  @Skip
```
Comment (robindenadel) [inline cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/DeviceManagement/DeviceLifeCycle.feature]: remove comment and @Skip

File: cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/DeviceManagement/DeviceLifeCycle.feature [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  Scenario Outline: Set dlms device status
+    Given an ssld oslp device
+      | DeviceIdentification | TEST1024000000001 |
+    When the SetDeviceLifecycleStatus request is received
+      | DeviceIdentification  | TEST1024000000001 |
+      | DeviceLifecycleStatus | <Status>          |
+    Then the device lifecycle status in the database is
+      | DeviceIdentification  | TEST1024000000001 |
+      | DeviceLifecycleStatus | <Status>          |
+    And the status change is logged in the audit trail
```
Comment (robindenadel) [inline cucumber-tests-platform-publiclighting/src/test/resources/features/publiclighting/osgp-adapter-ws-core/DeviceManagement/DeviceLifeCycle.feature]: remove this step

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/builders/entities/BaseDeviceBuilder.java [hunk 7]
```diff
@@ -37,7 +39,8 @@
     String gatewayDeviceIdentification = PlatformSmartmeteringDefaults.GATEWAY_DEVICE_IDENTIFICATION;
     Date technicalInstallationDate = PlatformSmartmeteringDefaults.TECHNICAL_INSTALLATION_DATE;
     DeviceModel deviceModel = PlatformSmartmeteringDefaults.DEVICE_MODEL;
-    boolean isActive = PlatformSmartmeteringDefaults.IS_ACTIVE;
+    // boolean isActive = PlatformSmartmeteringDefaults.IS_ACTIVE;
```
Comment (robindenadel) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/builders/entities/BaseDeviceBuilder.java]: remove this line

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/builders/entities/DeviceBuilder.java [hunk 8]
```diff
@@ -47,6 +42,10 @@ public Device build() {
         device.setVersion(this.version);
         device.setDeviceModel(this.deviceModel);
         device.setTechnicalInstallationDate(this.technicalInstallationDate);
+        // updateRegistrationDate sets the status to IN_USE, so setting of any
```
Comment (robindenadel) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/builders/entities/DeviceBuilder.java]: updateRegistrationDate -> updateRegistrationData

File: cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-core/DeviceLifeCycle.feature [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    Given a dlms device
+      | DeviceIdentification | TEST1024000000001 |
+      | DeviceType           | SMART_METER_E     |
+    When the SetDeviceLifecycleStatus request is received
+      | DeviceIdentification  | TEST1024000000001 |
+      | DeviceLifecycleStatus | <Status>          |
+    Then the device lifecycle status in the database is
+      | DeviceIdentification  | TEST1024000000001 |
+      | DeviceLifecycleStatus | <Status>          |
+    And the status change is logged in the audit trail
```
Comment (robindenadel) [inline cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-core/DeviceLifeCycle.feature]: remove this step

File: cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-core/DeviceLifeCycle.feature [hunk 10]
```diff
@@ -0,0 +1,31 @@
+@SmartMetering @Platform
+Feature: Core Operations, DeviceLifeCycle
+  As a grid operator
+  I want to distinguish the various statuses of a device
+  So I know what I can or cannot do with the device
+
+# This skip tag can be removed as soon as all code from shared and platform is merged and deployed.
+  @Skip
```
Comment (robindenadel) [inline cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-core/DeviceLifeCycle.feature]: remove comment and @Skip

File: cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-smartmetering/installation/AddMBusDevice.feature [hunk 11]
```diff
@@ -1,4 +1,4 @@
-@SmartMetering @Platform
+@SmartMetering @Platform @TEST
```
Comment (robindenadel) [inline cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-smartmetering/installation/AddMBusDevice.feature]: remove @TEST

File: cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-smartmetering/installation/AddSmartMeter.feature [hunk 12]
```diff
@@ -1,4 +1,4 @@
-@SmartMetering @Platform
+@SmartMetering @Platform @TEST
```
Comment (robindenadel) [inline cucumber-tests-platform-smartmetering/src/test/resources/features/osgp-adapter-ws-smartmetering/installation/AddSmartMeter.feature]: remove @TEST

File: cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/PlatformDefaults.java [hunk 13]
```diff
@@ -18,7 +19,8 @@
 public class PlatformDefaults {
 
     public static final Boolean DEFAULT_ACTIVATED = true;
-    public static final Boolean DEFAULT_ACTIVE = true;
+    // public static final Boolean DEFAULT_ACTIVE = true;
```
Comment (robindenadel) [inline cucumber-tests-platform/src/test/java/com/alliander/osgp/cucumber/platform/PlatformDefaults.java]: remove this line

**Comment by riannefijn** [comment]
retest this please

**Comment by villgust** [comment]
retest this please

**Pull request merged**